src/img/image.rs

```diff
@@ -32,6 +32,18 @@ impl Image {
         }
     }
 
+    // In these calculations are you computing with precision, but tossing
+    // the decimal portion when converting to usize? Just making sure that
+    // you want this to be the case:
+    // let a = 1000.59283191_f64;
+    // let b = a as usize;
+    // println!("a {}, b {}", a, b);
+    // stdout: a 1000.59283191, b 1000
+    // It makes sense to want an integer based on the API. But I also noticed
+    // you also convert from float to a rounded f32 when passing the value as
+    // an option to resize(). I don't understand what the intent is since it
+    // looks like options allows floats, but you only do integer math internally.
+    // Why support floats at all?
```

Comment:
I added support for floats late into developing this because I realized that clients using this service were sending floats for image dimensions (e.g. `?width=100.0`). I think there's probably updates I could make to support that in a cleaner way but I wasn't sure how to do that

Reply:
If you are intending on doing integer math only on the width and height, and you don't care about the fractional portion i.e. if user passes 10.123456789, you are fine with interpreting that as 10, then you can try taking in the query param as a float, but then immediately drop the fractional portion and only storing usize internally.